x-pack/elastic-agent/pkg/composable/providers/kubernetes/config.go

```diff
@@ -9,24 +9,62 @@ package kubernetes
 
 import (
 	"time"
+
+	"github.com/elastic/beats/v7/libbeat/logp"
 )
 
 // Config for kubernetes provider
 type Config struct {
+	Scope     string    `config:"scope"`
+	Resources Resources `config:"resources"`
+
 	KubeConfig     string        `config:"kube_config"`
+	Namespace      string        `config:"namespace"`
 	SyncPeriod     time.Duration `config:"sync_period"`
 	CleanupTimeout time.Duration `config:"cleanup_timeout" validate:"positive"`
 
-	// Needed when resource is a pod
+	// Needed when resource is a Pod or Node
 	Node string `config:"node"`
 
-	// Scope of the provider (cluster or node)
-	Scope string `config:"scope"`
+	LabelsDedot      bool `config:"labels.dedot"`
+	AnnotationsDedot bool `config:"annotations.dedot"`
```

Comment:
I'm not completely aware of pros/cons but with a quick view `flattened` type looks better than dedoting to me, and sounds like a good idea regarding timing to do this change now.

Reply:
taking this into account I'm inclined to leave dedotting out of this PR and investigate the experience when using flattened for these fields, any thoughts? Also, in case we end up introducing it, I would like to be opinionated here, and avoid adding any config parameter for it.

I'm particularly concerned about doing things like grouping some metric by a label, which is a valid use. I'm less concerned about annotations...

Reply:
I'm +1 in leaving this out for now and open a follow-up issue to work on for dedotting/flattening